### PR TITLE
Tests for the Shift Model

### PR DESCRIPTION
**Pull request by GabrielleEaston**
## What type of PR is this? (check all applicable)

- [] Feature
- [] Refactor
- [] Bug Fix
- [] Optimization
- [] Documentation Update
- [x] Other (describe: tests )

## Description of what PR does

- added tests to the Shift Model version-1

- Some tests are failing.

- I am not sure what is the best way to approach the testing of the private method

## Related PRs and/or Issues (if any)
- issue #6 
-

## QA Instructions, Screenshots, Recordings
run ``` bundle exec rspec spec/models/shift_spec.rb  ```

## Added tests?

- [x] yes
- [ ] no, because they aren't needed _(please include reasons for why tests aren't needed)_
- [ ] no, because I need help
- [ ] no, they will be added later (please create an Issue for it)

## Added to documentation?

- [ ] Yes, project README
- [x] No documentation needed

### CODE REVIEW

**Review by msarit — CHANGES_REQUESTED** [review]
GREAT first pass, Gabrielle! You're almost there!

I'm going to modify both the `shift.rb` (Shift model file) and `shifts.rb` (Shift factory); these changes will make some of your specs pass. Bravo - the way you wrote the specs on lines 12 and 14 led to these improvements in the code - which is the purpose of tests! 🎉

**Comment by GabrielleEaston** [comment]
@aritamana Thank you so much for reviewing! ❤️ I will continue working on it and make the requested changes tomorrow.

File: spec/models/shift_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  describe "filled_by(user) method" do
+    it "checks if worker who filled the shift is same as user" do
+    sample_user = create(:user, :worker_user_type)
+    sample_worker = create(:worker, user_id: sample_user.id)
+    sample_shift = create(:shift, :bartender_role, :filled_shift, worker_id: sample_worker.id)
+    expect(sample_worker.user_id === sample_user.id).to be(true)
+    end
+  end
+
+  describe "can_be_dropped?" do
```
[the remark this answers is not in the document]
Comment (msarit) [reply spec/models/shift_spec.rb:46]: @GabrielleEaston Looks like you overlooked this 😄

**Review by msarit — CHANGES_REQUESTED** [review]
Almost there! 😅  A few changes to make

File: spec/models/shift_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+RSpec.describe Shift, type: :model do
+  let(:shift1) { create(:shift, :bartender_role, :filled_shift) }
+  
+  
+  describe "validations" do
+    subject { shift1 }
+
+    it { is_expected.to validate_presence_of(:shift_role) }
+    it { is_expected.to validate_presence_of(:shift_description) }
+    it { is_expected.to validate_presence_of(:shift_start) }
```
[the remark this answers is not in the document]
Comment (GabrielleEaston) [reply spec/models/shift_spec.rb:12]: Thank you!

File: spec/models/shift_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      expect(sample_shift.shift_worker_name).to eq("#{sample_worker.first_name} #{sample_worker.last_name}")
+    end
+  end
+
+  describe "filled_by(user) method" do
+    it "should check if worker assigned to shift is same as current user" do
+    sample_user = create(:user, :worker_user_type)
+    sample_worker = create(:worker, user_id: sample_user.id)
+    sample_shift = create(:shift, :bartender_role, :filled_shift, worker_id: sample_worker.id)
+    expect(sample_worker.user_id === sample_user.id).to be(true)
```
[the remark this answers is not in the document]
Comment (msarit) [reply spec/models/shift_spec.rb:42]: @GabrielleEaston Hmmm I'm still seeing the old expectation 🤔 

I'm expecting (no pun intended) to see something like:

```ruby
expect(sample_shift.filled_by.....).to be(true)
```

Complete the expression inside the `expect()` 👍🏾

**Review by msarit — CHANGES_REQUESTED** [review]

**Review by msarit — APPROVED** [review]

**Pull request merged**